python/cog/server/worker.py

```diff
@@ -425,6 +427,25 @@ def _setup(self, redirector: AsyncStreamRedirector) -> None:
             # Could be a function or a class
             if hasattr(self._predictor, "setup"):
                 run_setup(self._predictor)
+
+            predict = get_predict(self._predictor)
+
+            is_async_predictor = inspect.iscoroutinefunction(
+                predict
+            ) or inspect.isasyncgenfunction(predict)
+
+            # Async models require python >= 3.11 so we can use asyncio.TaskGroup
+            # We should check for this before getting to this point
+            if is_async_predictor and sys.version_info < (3, 11):
+                raise FatalWorkerException(
+                    "Cog requires python >=3.11 for `async def predict(..)` support"
+                )
+
+            if self._max_concurrency > 1 and not is_async_predictor:
+                raise FatalWorkerException(
+                    "max_concurrency>1 requires `async def predict()`"
+                )
```

Comment:
Think we could describe `async def predict()` consistently either with or without `..`, please?